File: dev-docs/snippets/databases/subscribe.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Subscribe to real-time database changes for a server-registered subscription.
// JS-only — the Swift client has no equivalent.
export function subscribe(
  client: JsBaoClient,
  databaseId: string,
  subscriptionKey: string,
  currentUserId: string,
) {
  // #region example
  const unsub = client.databases.subscribe(databaseId, subscriptionKey, {
    params: { userId: currentUserId },
    onChange: (event) => {
      for (const change of event.changes) {
        console.log(change.op, change.id, change.data);
      }
    },
  });
  // Later: unsub();
  // #endregion example
  return unsub;
}

File: dev-docs/snippets/databases/subscribe.swift
import JsBaoClient

// Subscribe to real-time database changes for a server-registered subscription.
// Returns an unsubscribe handle — call it to stop receiving changes.
func subscribe(
  client: JsBaoClient,
  databaseId: String,
  subscriptionKey: String,
  currentUserId: String
) throws {
  // #region example
  let unsubscribe = try client.databases.subscribe(
    databaseId: databaseId,
    subscriptionKey: subscriptionKey,
    options: DatabaseSubscribeOptions(
      params: ["userId": currentUserId],
      onChange: { event in
        for change in event.changes {
          // change.changeType is "enter" | "update" | "leave"
          print(change.op, change.id, change.data ?? "")
        }
      }
    )
  )
  // Later: stop receiving changes.
  unsubscribe()
  // #endregion example
}
